ServiceApp/src/org/droidplanner/services/android/core/mission/survey/SurveyImpl.java

```diff
@@ -86,6 +86,10 @@ private void packSurveyPoints(List<msg_mission_item> list) {
         for (LatLong point : grid.gridPoints) {
             msg_mission_item mavMsg = getSurveyPoint(point, altitude);
             list.add(mavMsg);
+            if(surveyData.getLockOrientation()) {
+                mavMsg = getYawCondition(surveyData.getAngle());
```

Comment:
@TShapinsky can you use another variable to store the yaw condition here for improved readability.
Also rename the `mavMsg` variable above so that it's more descriptive.